### PR TITLE
KAFKA-6343 Documentation update in OS-level tuning section: add vm.ma…

### PR DESCRIPTION
**Pull request by rootex-**
Documentation update in OS-level tuning section: add vm.max_map_count bullet point

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by omkreddy** [comment]
LGTM

**Comment by afalko** [comment]
This suggestion was very useful for me in my dense partition environment. 👍

File: docs/ops.html [hunk 1]
```diff
@@ -624,10 +624,11 @@ <h4><a id="os" href="#os">OS</a></h4>
...
       <li>Max socket buffer size: can be increased to enable high-performance data transfer between data centers as <a href="http://www.psc.edu/index.php/networking/641-tcp-tune">described here</a>.
+	  <li>Maximum number of memory map areas a process may have (aka vm.max_map_count). <a href="http://kernel.org/doc/Documentation/sysctl/vm.txt">See the Linux kernel documentation</a>. You should keep an eye at this OS-level property when considering the maximum number of partitions a broker may have. By default, on a number of Linux systems, the value of vm.max_map_count is somewhere around 65535. Each new partition, hosted on broker, consumes 2 map areas. Thus, creating 50000 partitions on a broker will result allocation of 100000 map areas and likely cause broker crash with OutOfMemoryError (Map failed) on a system with default vm.max_map_count.
```
Comment (pxLi) [inline docs/ops.html]: I think its actually the index/timeindex consumes the mmaps, so partitions with large size or short log.roll.hours would use more. Not just the number of partitions~

Comment (tayler-jones) [reply docs/ops.html]: The index/timeindex do consume the mmaps, which makes it 2 per segment, not 2 per partition.

**Comment by ijuma** [comment]
Thanks for the PR. I think we probably need to update the file descriptor part as well since memory maps affect that too:

"The mmap() function adds an extra reference to the file associated with the file descriptor fildes which is not removed by a subsequent close() on that file descriptor. This reference is removed when there are no more mappings to the file." (thanks to @cmccabe for the reference)

http://pubs.opengroup.org/onlinepubs/7908799/xsh/mmap.html

cc @junrao

File: docs/ops.html (hunk 1 above)
Comment (rootex-) [reply docs/ops.html]: @pxLi  and @tayler-jones thanks for your remarks, please review the updates

**Comment by rootex-** [comment]
@ijuma, @pxLi, @tayler-jones 

Hello, friends! All corrections done, any additional notes maybe? Please let met know if there is something more we can do to improve this 😉

File: docs/ops.html [hunk 2]
```diff
@@ -668,10 +668,11 @@ <h4><a id="os" href="#os">OS</a></h4>
...
+	  <li>Maximum number of memory map areas a process may have (aka vm.max_map_count). <a href="http://kernel.org/doc/Documentation/sysctl/vm.txt">See the Linux kernel documentation</a>. You should keep an eye at this OS-level property when considering the maximum number of partitions a broker may have. By default, on a number of Linux systems, the value of vm.max_map_count is somewhere around 65535. Each log segment, allocated per partition, requires a pair of index/timeindex files, and each of these files consumes 1 map area. In other words, each log segment uses 2 map areas. Thus, each partition requires minimum 2 map areas, as long as it hosts a single log segment. That is to say, creating 50 000 partitions on a broker will result allocation of 100 000 map areas and likely cause broker crash with OutOfMemoryError (Map failed) on a system with default vm.max_map_count. Keep in mind that the number of log segments per partition varies depending on the segment size, load intensity, retention policy and, generally, tends to be more than one.
```
Comment (ijuma) [inline docs/ops.html]: Can you fix the indenting please?

Comment (rootex-) [reply docs/ops.html]: Ismael, done, thanks :ok_hand:

**Comment by bellemare** [comment]
Any chance on getting this merged? We ended up with a cluster failure due to vm.max_map_count issues, and only stumbled upon this PR after the fact. This would be helpful information for others. In addition, should we recommend a higher default? ElasticSearch recommends 262144 ( https://www.elastic.co/guide/en/elasticsearch/reference/current/vm-max-map-count.html ) and I have seen some Cassandra configs that are using 1M. What I cannot find is tradeoff-related information - what happens if vm.max_map_count  is set too high on the host.

File: docs/ops.html [hunk 3]
```diff
@@ -668,10 +668,11 @@ <h4><a id="os" href="#os">OS</a></h4>
...
+      <li>Maximum number of memory map areas a process may have (aka vm.max_map_count). <a href="http://kernel.org/doc/Documentation/sysctl/vm.txt">See the Linux kernel documentation</a>. You should keep an eye at this OS-level property when considering the maximum number of partitions a broker may have. By default, on a number of Linux systems, the value of vm.max_map_count is somewhere around 65535. Each log segment, allocated per partition, requires a pair of index/timeindex files, and each of these files consumes 1 map area. In other words, each log segment uses 2 map areas. Thus, each partition requires minimum 2 map areas, as long as it hosts a single log segment. That is to say, creating 50 000 partitions on a broker will result allocation of 100 000 map areas and likely cause broker crash with OutOfMemoryError (Map failed) on a system with default vm.max_map_count. Keep in mind that the number of log segments per partition varies depending on the segment size, load intensity, retention policy and, generally, tends to be more than one.
```
Comment (junrao) [inline docs/ops.html]: Could we remove the space in 50 000 and 100 000?

**Review by junrao — APPROVED** [review]
@rootex- : Thanks for the patch. LGTM. Just a minor comment below.

File: docs/ops.html (hunk 3 above)
Comment (rootex-) [reply docs/ops.html]: Sure. Done. I thought it added some readability though...

**Comment by rootex-** [comment]
Any more corrections/suggestions?

**Review by junrao — APPROVED** [review]
@rootex- : Thanks for the updated patch. LGTM. Sorry for the delay.

**Comment by rootex-** [comment]
@junrao Thanks, good to know it's a part of documentation now, thus, helpful to others👍🏽

**Pull request merged**